File: backend/routers/community.py
from fastapi import APIRouter, HTTPException
from typing import List
from models import CommunityPost
from datetime import datetime

router = APIRouter()

# Mock data
mock_posts = [
    CommunityPost(
        id=1,
        title="Best strategies for Season 2?",
        content="What are your top strategies for the new season?",
        author="GameEnthusiast",
        created_at=datetime(2025, 10, 21),
        replies=23,
        likes=45
    )
]

@router.get("/board", response_model=List[CommunityPost])
async def get_board_posts():
    """Get community board posts"""
    return mock_posts

@router.get("/board/{post_id}", response_model=CommunityPost)
async def get_post(post_id: int):
    """Get a specific post by ID"""
    for post in mock_posts:
        if post.id == post_id:
            return post
    raise HTTPException(status_code=404, detail="Post not found")

@router.get("/links")
async def get_community_links():
    """Get community links"""
    return {
        "official": [
            {"name": "Official Website", "url": "https://legends.dalarnia.com/"},
            {"name": "Discord", "url": "#"},
            {"name": "Twitter", "url": "#"}
        ],
        "community": [
            {"name": "Reddit", "url": "#"},
            {"name": "YouTube", "url": "#"}
        ]
    }

@router.get("/contacts")
async def get_contacts():
    """Get contact information"""
    return {
        "email": "contact@dalarnia.fun",
        "discord": "dalarnia-legends",
        "social_media": {
            "twitter": "@DalarniaLegends",
            "facebook": "DalarniaLegends"
        }
    }

File: frontend/src/data/community.ts
import { CommunityLinks, CommunityPost, ContactInfo } from "@/types";

export const communityPosts: CommunityPost[] = [
  {
    id: "post-1",
    title: "What counters Rift Tempo?",
    content:
      "I keep running into Lyra tempo decks in platinum. Any tech picks that slow them without giving up late game?",
    author: "MinerMike",
    replies: 12,
    likes: 48,
    createdAt: "2024-03-09T11:00:00.000Z",
  },
  {
    id: "post-2",
    title: "Share your Mining Rush drops",
    content:
      "Pulled a radiant relic on my third run! Curious what everyone else found and whether the drop table is worth grinding.",
    author: "CrystalQueen",
    replies: 34,
    likes: 72,
    createdAt: "2024-03-07T15:45:00.000Z",
  },
  {
    id: "post-3",
    title: "LFG for Riftbreakers Championship",
    content:
      "Looking for two consistent teammates (supports preferred) to queue for the qualifier. We scrim daily at 20:00 UTC.",
    author: "Echelon",
    replies: 9,
    likes: 33,
    createdAt: "2024-03-05T19:30:00.000Z",
  },
];

export const communityLinks: CommunityLinks = {
  official: [
    { name: "PlayDalarnia.com", url: "https://playdalarnia.com" },
    { name: "Status Page", url: "https://status.dalarnia.fun" },
    { name: "Official Discord", url: "https://discord.gg/dalarnia" },
  ],
  community: [
    { name: "Reddit /r/DalarniaLegends", url: "https://reddit.com/r/DalarniaLegends" },
    { name: "Strategy Hub Notion", url: "https://notion.so/dalarnia-legends" },
    { name: "Esports Bracket", url: "https://battlefy.com/dalarnia-legends" },
  ],
};

export const contactInfo: ContactInfo = {
  email: "contact@dalarnia.fun",
  discord: "dalarnia-legends",
  socialMedia: {
    twitter: "@DalarniaLegends",
    facebook: "DalarniaLegends",
  },
};
